Feature: Add AutoDrive Simulator Dockerfile and Adds AutoDrive Devkit to robocar racestack's docker image

chenyenru opened the pull request:
Regarding Dockerfile:

- Could we name Dockerfile as `<name>.Dockerfile`? This could improve readability and follow Docker's naming convention, see https://docs.docker.com/build/concepts/dockerfile/#:~:text=Some%20projects%20may,%3Csomething%3E.Dockerfile
- Should the file be a replacement of `Dockerfile.f1tenth_x86` instead of a separate Dockerfile?

chenyenru committed:
feat: Added AutoDrive Simulator Dockerfile and entry point script

Helpful when developing racestack with AutoDrive Simulator

chenyenru committed:
feat: Merge f1tenth_x86 and AutoDrive Devkit Dockerfiles

Introduces a new Dockerfile that combines the dependencies and configurations from both the f1tenth_x86 and AutoDrive Devkit Dockerfiles. The resulting Docker container will include all necessary dependencies to run the AutoDrive Devkit. Additionally, the AutoDrive Devkit repository will be cloned into the `src` directory within the container.

Source of AutoDrive Devkit: https://github.com/AutoDRIVE-Ecosystem/AutoDRIVE-F1TENTH-Sim-Racing/releases/tag/2024-cdc

WinstonHChou commented on autodrive_devkit.sh:
```diff
@@ -0,0 +1,15 @@
+
+# Setup development environment
+source /opt/ros/foxy/setup.bash
+source /home/projects/f1tenth_ws/autodrive_devkit/install/setup.bash
```
Please put `autodrive_devkit/` under ` src/external/f1tenth_sim/autodrive_devkit/`
It should not be tracked by git

WinstonHChou commented on .docker_utils/AutoDriveF1Tenth_x86.Dockerfile:
```diff
@@ -0,0 +1,180 @@
+
+# Copy and install AutoDRIVE Devkit (ROS 2 API)
+RUN wget https://github.com/AutoDRIVE-Ecosystem/AutoDRIVE-F1TENTH-Sim-Racing/releases/download/2024-cdc/autodrive_devkit.zip && \
+    unzip autodrive_devkit.zip -d ${WORKSPACE}src/ && \
```
Please put `autodrive_devkit/` under ` src/external/f1tenth_sim/autodrive_devkit/`
It should not be tracked by git

WinstonHChou reviewed (changes requested):
Please put `autodrive_devkit/` under ` src/external/f1tenth_sim/autodrive_devkit/`

chenyenru committed:
refactor: Integrate AutoDrive Devkit support into `f1tenth_x86` image

chenyenru committed:
feat: Add run script for autodrive simulator in docker container

chenyenru committed:
feat: Add autodrive devkit launch file in basestation_launch to support autodrive simulator in robocar racestack

chenyenru committed:
feat: Added ackermann_odom dependency and  ackermann_odom_node launching in autodrive_sim.launch.py

WinstonHChou commented on repos/autodrive.repos:
```diff
@@ -0,0 +1,6 @@
+repositories:
+    src/external/ackermann_odom:
```
change the directory to `src/external/actuator/ackermann_odom`
Please add annotation in https://github.com/chenyenru/ackermann-odom-ros2.git 
If you used code from other organization

WinstonHChou replied:
```
Starting >>> ackermann_odom
--- stderr: ackermann_odom                           
CMake Error at CMakeLists.txt:17 (find_package):
  By not providing "Findbackward_ros.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "backward_ros", but CMake did not find one.

  Could not find a package configuration file provided by "backward_ros" with
  any of the following names:

    backward_rosConfig.cmake
    backward_ros-config.cmake

  Add the installation prefix of "backward_ros" to CMAKE_PREFIX_PATH or set
  "backward_ros_DIR" to a directory containing one of the above files.  If
  "backward_ros" provides a separate development package or SDK, be sure it
  has been installed.


---
Failed   <<< ackermann_odom [17.6s, exited with code 1]
```

WinstonHChou replied:
please remove this dependency if you dont need it
I'd prefer not to install extra dependencies

WinstonHChou replied:
```
[ERROR] [ackermann_odom_node-4]: exception occurred while executing process:
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 723, in __execute_process
    transport, self._subprocess_protocol = await async_execute_process(
  File "/opt/ros/foxy/lib/python3.8/site-packages/osrf_pycommon/process_utils/async_execute_process_asyncio/impl.py", line 142, in async_execute_process
    transport, protocol = yield from _async_execute_process_nopty(
  File "/opt/ros/foxy/lib/python3.8/site-packages/osrf_pycommon/process_utils/async_execute_process_asyncio/impl.py", line 46, in _async_execute_process_nopty
    transport, protocol = yield from loop.subprocess_exec(
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1630, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "/usr/lib/python3.8/asyncio/unix_events.py", line 197, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,
  File "/usr/lib/python3.8/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
  File "/usr/lib/python3.8/asyncio/unix_events.py", line 789, in _start
    self._proc = subprocess.Popen(
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gdb'
```

WinstonHChou commented on src/launch/basestation_launch/param/autodrive_sim.yaml:
you are missing `map_path: "<map_path>"` for yaml configuration

chenyenru committed:
[Feature] vcs import ackermann odom in the docker container

chenyenru committed:
[Refactoring] Temporarily commenting out nav2 nodes in autodrive_sim.launch.py

chenyenru committed:
[BugFix] Update Dockerfile for vcs import and update autodrive.repos with the correct URL

chenyenru committed:
[BugFix] Resolve missing livox dependency for fast lio

Moved` make livox-driver` before `make racer` so that `livox-driver` is there as a dependency for `fast lio`

chenyenru committed:
[Refactoring] Remove autodrive simulator and dockerfile since we'll be pulling image from docker hub

chenyenru committed:
[Refactor] Remove autodrive_simulator.sh

WinstonHChou reviewed (approved):
ok
